src/reconcile/rec_visibility.c

```diff
@@ -417,12 +417,52 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
 
             /* Find the update this tombstone applies to. */
             if (!__wt_txn_upd_visible_all(session, upd)) {
-                while (upd->next != NULL && upd->next->txnid == WT_TXN_ABORTED)
-                    upd = upd->next;
+                /*
+                 * Loop until a valid update from a different transaction is found in the update
+                 * list.
+                 */
+                while (upd->next != NULL) {
+                    if (upd->next->txnid == WT_TXN_ABORTED)
+                        upd = upd->next;
+                    else if (tombstone->txnid == upd->next->txnid &&
+                      tombstone->start_ts == upd->next->start_ts &&
```

Comment:
This doesn't handle the case where multiple timestamps are used in the same transaction. I think no need to compare the timestamps, comparing the txnid will suffice.

Reply:
Fixed by comparing only the transaction id. Also added a check that the transaction id must not be WT_TS_NONE. The updates that are retrieved by RTS don't have transaction id after the restart.